tests/commands/test_checks.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import pytest
+
+from django.core.management import call_command
+from django.core.management.base import CommandError
+from django.db import connection
+
+
+@pytest.mark.cmd
+def test_test_checks_noargs():
+    """No args should fail wanting either --unit or --source."""
+    with pytest.raises(CommandError) as e:
+        call_command('test_checks')
+    assert ("Either --unit or a pair of --source and "
+            "--target must be provided." in str(e))
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_test_checks_unit_unkown(afrikaans_tutorial):
+    """Check a --unit that we won't have"""
+    with pytest.raises(CommandError) as e:
+        call_command('test_checks', '--unit=100000')
+    assert 'Unit matching query does not exist' in str(e)
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_test_checks_unit(capfd, site_matrix):
+    """Check a --unit"""
+    # Unit 3/104 is /language0/project0/store0.po Translated Target
+    # unitid varies on different databases
+    if connection.vendor == 'sqlite':
+        unitid = 196
+    else:
+        unitid = 3
+    call_command('test_checks', '--unit=%s' % unitid)
+    out, err = capfd.readouterr()
+    assert 'No errors found' in out
+
+
+@pytest.mark.cmd
+def test_test_checks_srctgt_missing_args():
+    """Check a --source --target with incomplete options."""
+    with pytest.raises(CommandError) as e:
+        call_command('test_checks', '--source="%s files')
+    assert 'Use a pair of --source and --target' in str(e)
+    with pytest.raises(CommandError) as e:
+        call_command('test_checks', '--target="%s leers')
```

Comment:
I think that using the `%s` placeholder in the test string might confuse anyone reading the code. I suggest using just `Files` and `Leers`.